src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -28,17 +28,26 @@ internal static partial class FileSystem
             UnixFileMode.OtherWrite |
             UnixFileMode.OtherExecute;
 
+        // TryCloneFile is defined in either FileSystem.TryCloneFile.OSX.cs or FileSystem.TryCloneFile.OtherUnix.cs.
+        private static partial bool TryCloneFile(string sourceFullPath, in Interop.Sys.FileStatus srcStat, string destFullPath, bool overwrite);
+
         public static void CopyFile(string sourceFullPath, string destFullPath, bool overwrite)
         {
-            long fileLength;
-            UnixFileMode filePermissions;
-            using SafeFileHandle src = SafeFileHandle.OpenReadOnly(sourceFullPath, FileOptions.None, out fileLength, out filePermissions);
-            using SafeFileHandle dst = SafeFileHandle.Open(destFullPath, overwrite ? FileMode.Create : FileMode.CreateNew,
-                                            FileAccess.ReadWrite, FileShare.None, FileOptions.None, preallocationSize: 0, filePermissions,
-                                            CreateOpenException);
+            // Open the src file handle.
+            using SafeFileHandle src = SafeFileHandle.OpenReadOnly(sourceFullPath, FileOptions.None, out Interop.Sys.FileStatus srcFileStatus);
```

Comment:
https://github.com/dotnet/runtime/pull/79243#discussion_r1197320789 is not yet resolved.

Reply:
The funny thing is that I did fix it, but then it got unfixed when I was copying code around :)

Reply:
This function should only have changes for adding the `TryCloneFile` call. The remainder should remain as it was.

Reply:
It can't be the same, SafeFileHandle.OpenReadOnly had to change for example. I'll try to make it more similar though.